examples/Hdmx/main.c

```diff
@@ -288,107 +292,113 @@ int main (int argc, char **argv)
     servaddr.sin_port = htons(ARTNET_PORT); 
       
     // Bind the socket with the server address 
-    if ( bind(sockfd, (const struct sockaddr *)&servaddr,  
+    if ( bind(artnetSock, (const struct sockaddr *)&servaddr,  
             sizeof(servaddr)) < 0 ) 
     { 
         perror("bind failed"); 
         exit(EXIT_FAILURE); 
     } 
 
+    struct timeval tv;
+    tv.tv_sec = 5;
+    tv.tv_usec = 0;
+    setsockopt(artnetSock, SOL_SOCKET, SO_RCVTIMEO,&tv,sizeof(tv));
+
 
    unsigned char sequence;
     while(1)
 	    {
 	    int upd_len = sizeof(cliaddr);  //len is value/resuslt 
 	  
-	    int n = recvfrom(sockfd, (char *)buffer, MAX_ARTNET_SIZE,  
-			MSG_WAITALL, ( struct sockaddr *) &cliaddr, 
+	    int n = recvfrom(artnetSock, (char *)buffer, MAX_ARTNET_SIZE,  
+			0, ( struct sockaddr *) &cliaddr, 
 			&upd_len); 
-
-
-	    //Magic Number
-	    if(buffer[7]!=0)
-		continue;
-	    if(strcmp(buffer,"Art-Net")!=0)
-		continue;
-
-	    //Opcode
-            int opCodeHandled;
-            if(buffer[8]!=0)
+	    if(n>10)
 		{
-		printf("Bad Opcode\n");
-		continue;
-		}
-
-	    switch(buffer[9])
-		{
-	        case 0x20:
+		    //Magic Number
+		    if(buffer[7]!=0 || strcmp(buffer,"Art-Net")!=0)
 			{
-			if(debug_level>=MSG_INFO)
-				{
-				printf("Poll:");
-				printf("%s:",inet_ntoa(cliaddr.sin_addr));
-				printf("%d\n",ntohs(cliaddr.sin_port));
-				}
-			sendPollReply(&cliaddr,sockfd);
-			break;
+			printf("Bad Magic\n");
+			continue;
 			}
-	        case 0x80:
-			{
-			if(debug_level>=MSG_INFO)
-				printf("TodRequest\n");
 
-			break;;
-			}
-	        case 0x21:
+		    //Opcode
+		    int opCodeHandled;
+		    if(buffer[8]!=0)
 			{
-			if(debug_level>=MSG_INFO)
-				{
-				printf("Poll Reply\n");
-				printf("%s\n",buffer+26);
-				printf("%s\n",buffer+44);
-				printf("%s\n",buffer+108);
-				break;
-				}
+			printf("Bad Opcode\n");
+			continue;
 			}
 
-	        case 0x50:
+		    switch(buffer[9])
 			{
-			//Protocaol (14)
-			 if(buffer[10]!=0)
-				break;
-			 if(buffer[11]!=14)
+			case ARTNET_OP_POLL:
+				{
+				if(debug_level>=MSG_INFO)
+					{
+					printf("Poll:");
+					printf("%s:",inet_ntoa(cliaddr.sin_addr));
+					printf("%d\n",ntohs(cliaddr.sin_port));
+					}
+				sendPollReply(&cliaddr,artnetSock);
 				break;
+				}
+			case ARTNET_OP_TOD_REQUEST:
```

Comment:
I'm getting a `main.c:346:4: warning: case label value exceeds maximum value for type` compiler warning on this line... should `buffer` be defined as `unsigned char` instead of `char`?